rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1829,6 +1832,82 @@ public Response createStudentContactLogEntry(@PathParam("ID") Long id, fi.otavan
   }
 
+  @Path("/students/contactLogEntries/batch")
+  @POST
+  @RESTPermit(handling = Handling.INLINE)
+  public Response createMultipleStudentContactLogEntries(fi.otavanopisto.pyramus.rest.model.StudentContactLogWithRecipients entity) {
+    if (entity == null) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+    
+    if (!restSecurity.hasPermission(new String[] { StudentPermissions.FIND_STUDENT })) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    fi.otavanopisto.pyramus.rest.model.StudentContactLogWithRecipients cLE = null;
+    
+    if (entity.getRecipients() != null) {
+      
+      List<Long> recipients = new ArrayList<>();
+      
+      User loggedUser = sessionController.getUser();
+      StaffMember creator = userController.findStaffMemberById(loggedUser.getId());
+      
+      StudentContactLogEntryType type = entity.getType() != null ? StudentContactLogEntryType.valueOf(entity.getType().name()) : null;
+      StudentContactLogEntry contactLogEntry = null;
+      for (Long recipient : entity.getRecipients()) {
+        Student student = studentController.findStudentById(recipient);
+        
+        Status studentStatus = checkStudent(student);
+        if (studentStatus != Status.OK)
+          return Response.status(studentStatus).build();
+        
+        if (student == null) {
+          StaffMember staff = userController.findStaffMemberById(recipient);
+
+          if (staff != null) {
+            continue;
+          }
+          return Response.status(Status.BAD_REQUEST).build();
+        }
+        
+        recipients.add(recipient);
+        
+        ContactLogAccess access = studentController.resolveContactLogAccess(student);
+        
+        if(!access.equals(ContactLogAccess.ALL) && !access.equals(ContactLogAccess.OWN)) {
+          return Response.status(Status.FORBIDDEN).build();
+        }
+        
+        contactLogEntry = studentContactLogEntryController.createContactLogEntry(student, type, entity.getText(),
+            toDate(entity.getEntryDate()), loggedUser.getFullName(), creator);
+
+      
+      }
+      fi.otavanopisto.pyramus.rest.model.StudentContactLogEntryType entryType = fi.otavanopisto.pyramus.rest.model.StudentContactLogEntryType.valueOf(contactLogEntry.getType().name());
+      
+      Date tmpDate = new Date(contactLogEntry.getEntryDate().getTime()); 
+      Instant instant = tmpDate.toInstant();
+      ZoneId systemId = ZoneId.systemDefault();
+      ZoneOffset offset = systemId.getRules().getOffset(instant);
+      OffsetDateTime entryDate = tmpDate.toInstant().atOffset(offset);
+      
+      Long creatorId = contactLogEntry.getCreator() != null ? contactLogEntry.getCreator().getId() : null;
+
+      cLE = new fi.otavanopisto.pyramus.rest.model.StudentContactLogWithRecipients();
+      cLE.setId(contactLogEntry.getId());
+      cLE.setType(entryType);
+      cLE.setText(contactLogEntry.getText());
+      cLE.setEntryDate(entryDate);
```

Comment:
I suppose the shortest Date to OffsetDateTime would be something like...

`OffsetDateTime.ofInstant(contactLogEntry.getEntryDate().toInstant(), ZoneId.systemDefault())`